app/api/server/lib/integrations.js

```diff
@@ -0,0 +1,25 @@
+import { Integrations } from '../../../models/server/raw';
+import { hasPermissionAsync } from '../../../authorization/server/functions/hasPermission';
+
+const hasIntegrationsPermission = async (userId, integration) => {
+	if (!integration) {
+		throw new Error('integration is required');
+	}
+	const canManageIncomingIntegrations = await hasPermissionAsync(userId, 'manage-incoming-integrations') || (userId === integration._createdBy._id && await hasPermissionAsync(userId, 'manage-own-incoming-integrations'));
```

Comment:
checking permissions can be a heavy task.. so I think this could be changed to something like:
```javascript
const type = integration.type === 'webhook-incoming' ? 'incoming' : 'outgoing';

if (await hasPermissionAsync(userId, `manage-${ type }-integrations`)) {
	return true;
}

if (userId === integration._createdBy._id) {
	return hasPermissionAsync(userId, `manage-own-${ type }-integrations`);
}

return false;
```

this way it will only check the needed permissions